src/Compilation.zig

```diff
@@ -124,7 +124,7 @@ fn generateDateAndTime(w: anytype) !void {
 
     const day_names = [_][]const u8{ "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun" };
     // days since Thu Oct 1 1970
-    const day_name = day_names[@truncate(u8, (epoch_day.day + 3) % 7)];
+    const day_name = day_names[@as(u8, @truncate((epoch_day.day + 3) % 7))];
```

Comment:
Is this `@as` required?

Reply:
Removed the cast and the truncate

Reply:
Seems like the truncate is needed for 32-bit builds.

Reply:
The `@as` should still be unnecessary.

Reply:
Without it, on a 64bit system I get

```
src/Compilation.zig:127:32: error: destination type 'usize' has more bits than source type 'u47'
    const day_name = day_names[@truncate((epoch_day.day + 3) % 7)];
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/Compilation.zig:127:32: note: destination type has 64 bits
src/Compilation.zig:127:62: note: operand type has 47 bits
    const day_name = day_names[@truncate((epoch_day.day + 3) % 7)];
                                         ~~~~~~~~~~~~~~~~~~~~^~~
```

Reply:
Maybe we should just be using `@intCast` in the first place instead of truncate